src/Nevermind/Nevermind.Evm/ITransactionStore.cs

```diff
@@ -0,0 +1,16 @@
+using Nevermind.Core;
+using Nevermind.Core.Crypto;
+
+namespace Nevermind.Evm
+{
+    public interface ITransactionStore
+    {
+        void AddTransaction(Transaction transaction);
+        void AddTransactionReceipt(Keccak transactionHash, TransactionReceipt transactionReceipt, Keccak blockhash);
+        Transaction GetTransaction(Keccak transactionHash);
+        TransactionReceipt GetTransactionReceipt(Keccak transactionHash);
+        bool WasProcessed(Keccak transactionHash);
```

Comment:
what does it mean? it may not be on the main chain even if it was added, maybe worth to clarify the API, let us discuss